remove ForceLogging flag, always use StartCluster

tbg opened the pull request:
only StartCluster sets up the logging directory for
per-test usage.
ForceLogging was made obsolete by changing `util/log`
to return an empty result instead of an error when
no log dirs were defined.

<!-- Reviewable:start -->

[<img src="https://reviewable.io/review_button.png" height=40 alt="Review on Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/3924)

<!-- Reviewable:end -->

tbg committed:
remove ForceLogging flag, always use StartCluster

only StartCluster sets up the logging directory for
per-test usage.
ForceLogging was made obsolete by changing `util/log`
to return an empty result instead of an error when
no log dirs were defined.

tamird commented:
Reviewed 6 of 6 files at r1.
Review status: all files reviewed at latest revision, 2 unresolved discussions.

---

*[acceptance/main_test.go, line 141 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/3924#-K8TzkFhch6QxA2NlBvJ:-K8TzkFiBzDPHLXubwIE:-663709510) ([raw file](https://github.com/cockroachdb/cockroach/blob/cf4bec56a32f1b36b1a0775d67230cea06c83ea2/acceptance/main_test.go#L141)):*
nit: s/skips/calls t.Skip/

---

*[acceptance/status_server_test.go, line 98 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/3924#-K8Tzo_pb4o2D4Mv9WT9:-K8Tzo_pb4o2D4Mv9WTA:-831634568) ([raw file](https://github.com/cockroachdb/cockroach/blob/cf4bec56a32f1b36b1a0775d67230cea06c83ea2/acceptance/status_server_test.go#L98)):*
how come this type assertion is always done? could we change the signature instead?

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/cockroachdb/cockroach/3924)*
<!-- Sent from Reviewable.io -->

bdarnell commented:
LGTM. Can you add a comment to each local-only test about why it's local-only? It's relatively obvious for the docker test but I'm not sure why some of the others are.

tbg committed:
enable all but one test for remote clusters

disclaimer: haven't checked that they pass for remote clusters.
but we're not running these regularly, and if they should fail
it's an indication to fix either the remote cluster setup or
something inside of cockroach proper.

tschottdorf commented:
I just went and fixed up all the tests that were using a local cluster (except the explicit docker one). PTAL

---

Review status: 2 of 6 files reviewed at latest revision, all discussions resolved.

---



*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/cockroachdb/cockroach/3924#-:-K8WcmHDYh_04GaHqmf7:1950639659)*
<!-- Sent from Reviewable.io -->

tamird commented:
Reviewed 4 of 4 files at r2.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/cockroachdb/cockroach/3924)*
<!-- Sent from Reviewable.io -->

bdarnell commented:
LGTM
